crates/bitwarden/src/auth/login/access_token.rs

```diff
@@ -82,12 +110,43 @@ async fn request_access_token(
         .await
 }
 
+fn load_tokens_from_state(
+    client: &mut Client,
+    state_file: &Path,
+    access_token: &AccessToken,
+) -> Result<Uuid> {
+    let client_state = state::get(state_file, access_token)?;
+
+    if let Some(client_state) = client_state {
+        let token: JWTToken = client_state.token.parse()?;
+
+        if let Some(organization_id) = token.organization {
+            let time_till_expiration = (token.exp as i64) - Utc::now().timestamp();
+
+            if time_till_expiration > 0 {
+                let organization_id: Uuid = organization_id
+                    .parse()
+                    .map_err(|_| Error::Internal("Bad organization id."))?;
+                let encryption_key: SymmetricCryptoKey = client_state.encryption_key.parse()?;
+
+                client.set_tokens(client_state.token, None, time_till_expiration as u64);
+                client.initialize_crypto_single_key(encryption_key);
+
+                return Ok(organization_id);
+            }
+        }
+    }
+
+    Err(Error::Internal("Could not load tokens from state."))
```

Comment:
Do we want some better error handling here? To improve debug experience.

Reply:
Yes, we can definitely do better than using `Internal`. Added a new `InvalidStateFile` variant to our `Error` enum. Resolved in [d0c0c10](https://github.com/bitwarden/sdk/pull/388/commits/d0c0c10dadd1a72550a2eccdd861b1a44c8f1859)